dotnet/src/Connectors/Connectors.Memory.Redis/RedisJsonDynamicDataModelMapper.cs

```diff
@@ -59,28 +60,23 @@ public VectorStoreGenericDataModel<string> MapFromStorageToDataModel((string Key
             _ => throw new VectorStoreRecordMappingException($"Invalid data format for document with key '{storageModel.Key}'"),
         };
 
-        foreach (var property in properties)
+        foreach (var property in model.Properties)
         {
-            var targetDictionary = property is VectorStoreRecordDataPropertyModel ? dataModel.Data : dataModel.Vectors;
-
-            // Only map properties across that actually exist in the input.
-            if (!jsonObject.TryGetPropertyValue(property.StorageName, out var sourceValue))
+            // Key handled above
+            if (property is VectorStoreRecordKeyPropertyModel)
             {
                 continue;
             }
 
-            // Replicate null if the property exists but is null.
-            if (sourceValue is null)
+            // Only map properties across that actually exist in the input.
```

Comment:
Is this consistent with the other mappers? It seemed like we were now about to always express nulls in explicit way (by adding a null value to the dictionary)

Reply:
We're not very consistent around this, no... I already have a note to discuss this with you guys at some point and arrive at a desired behavior...

In the meantime, this PR should generally not change the behavior - and I think that's the case no? If the property is missing, we skip it (don't populate null), otherwise we populate.